### DISCUSSION TITLE
MSelect控件绑定多选遇到问题，SelectedItems数据无法取到，望解答

### DISCUSSION THREAD

**Discussion by LZD121** [BlazorComponent/MASA.Blazor#484] [Q&A] [answered]
背景：我在项目中尝试用模板的方式生成所有的表单控件：
例如：
![image](https://user-images.githubusercontent.com/59716053/170407720-e473bc57-f150-4d20-a596-a61a85f09f72.png)

但是在使用多项选择时遇到问题：
![image](https://user-images.githubusercontent.com/59716053/170407957-ec7f4389-c13c-4b22-8690-d1c4e508d46e.png)

我把用来绑定多项列表的List存入了一个字典“_nestDataDictionary”，字典的目的是用来模板化动态存储不同名的字段所绑定的数据。
在MSelect 的@bind-value上设置为 “@(_nestDataDictionary[dataProperty.Name])”
此时启动程序多选框能能正确识别到字典中对应的key下的List并显示出来，
但是当我对选项进行了修改，添加几个选项并点击保存的时候，这个字典中的List中的数据没有变化，我添加的对象没有被添加到这个List中，Debug发现这些项目只存在MSelect的SelectedItems里面，但是控件没有提供公开的方法用来获取这些数据？

请问这个问题如何解决？？

另外我原本的方案是使用MSelect上的Value和ValueChanges事件来操作，而不是使用@bind-value:
![image](https://user-images.githubusercontent.com/59716053/170413347-d89337dd-dd5a-4f95-a84b-fbff443cac6f.png)

这样我可以自己控制数据变更后设置的方式。
但这个方式的问题是Value会把设置的List列表识别为一个object对象（由于是使用模板的方式进行生成，所以无法在编译时声明数据的类型）
而当ValueChanged事件触发的时候接收到的参数会呈现为
{
List<object>,
object,
object,
}
也就是控件不把设置的对象识别为一个容器，并且在这种方式下界面上无法呈现List<object>中已有的选项。

**Comment by doddgu** [comment]
cc @capdiem

**Comment by capdiem** [comment, marked as answer]
@LZD121  能不能提供一个最小的demo或repository？

**Reply by LZD121** [reply]
没有问题了，调试出来是组件多次被呈现所以代码也跑了多遍导致字典里指向错的对象了。
看来这种方法还要考虑组件刷新的问题。。。
